src/index/updater/inscription_updater/stream.rs

```diff
@@ -0,0 +1,474 @@
+    self.metaprotocol = inscription.metaprotocol().map(|mp| mp.to_owned());
+    self.metadata = inscription.metadata();
+    self.pointer = inscription.pointer();
+    self.content_type = inscription
+      .content_type()
+      .map(|content_type| content_type.to_string());
+    self.content_length = inscription.content_length();
+    self.content_media = Some(inscription.media().to_string());
+    self.content_body = match inscription.body() {
+      Some(body) => {
+        // only encode if the body length is less than 1M bytes
+        let kafka_body_max_bytes = env::var("KAFKA_BODY_MAX_BYTES")
+          .unwrap_or("950000".to_owned())
+          .parse::<usize>()
+          .unwrap();
+
+        if Self::is_text_related(inscription.media(), self.content_type.clone())
+          && body.len() < kafka_body_max_bytes
+        {
+          self.brc20 = serde_json::from_slice(body).unwrap_or(None);
+          self.domain = Domain::parse(body);
+          Some(general_purpose::STANDARD.encode(body))
+        } else {
+          None
+        }
+      }
+      None => None,
+    };
+    self.delegate = inscription.delegate().map(|d| d.to_string());
+    self
+  }
+
+  pub(crate) fn with_transfer(&mut self, old_satpoint: SatPoint, index: &Index) -> &mut Self {
+    self.old_location = Some(old_satpoint);
+
+    self.old_owner = OLD_OWNER_CACHE
+      .get(&old_satpoint.outpoint)
+      .unwrap_or_else(|| {
+        let old_owner = index
+          .get_transaction(old_satpoint.outpoint.txid)
+          .unwrap_or(None)
+          .and_then(|tx| {
+            tx.output
+              .get(old_satpoint.outpoint.vout as usize)
+              .and_then(|txout| {
+                Address::from_script(&txout.script_pubkey, StreamEvent::get_network())
+                  .map_err(|e| {
+                    warn!(
+                      "StreamEvent::with_transfer could not parse old_owner address: {}",
+                      e
+                    );
+                  })
+                  .ok()
+              })
+          });
+        OLD_OWNER_CACHE.set(old_satpoint.outpoint, old_owner.clone());
+        old_owner
+      });
+
+    // Only enrich the inscription if it is a BRC20 transfer
+    if *IS_BRC20 {
+      match index
+        .get_inscription_by_id_unsafe(self.inscription_id)
+        .unwrap_or(None)
+      {
+        Some(inscription) => {
+          self.enrich_content(inscription);
+        }
+        None => {
+          warn!("could not find inscription for id {}", self.inscription_id);
+        }
+      }
+    }
+    self
+  }
+
+  pub(crate) fn with_create(
+    &mut self,
+    sat: Option<Sat>,
+    inscription_number: i64,
+    inscription: Inscription,
+    parents: Vec<InscriptionId>,
+    charms: u16,
+  ) -> &mut Self {
+    self.enrich_content(inscription);
+    self.sat = sat;
+    self.inscription_number = Some(inscription_number);
+    self.parent = parents.first().copied();
```

Comment:
this is the main change, multi-parent inscriptions are supported in upstream but our messages only support single parent right now